### ISSUE TITLE
Unable to install package vibe-d using dub package manager for d language.

### ISSUE THREAD

**Issue by Fernal73** [termux/termux-packages#7089]
**Problem description**
Unable to install package vibe-d using d language package manager dub.

**Steps to reproduce**

$ dub run vibe-d -v  &> vibe.out

$ cat vibe.out | termux-clipboard-set

(Paste as below)

```d
Using dub registry url 'https://code.dlang.org/'
Refreshing local packages (refresh existing: true)...
Looking for local package map at /var/lib/dub/packages/local-packages.json
Looking for local package map at /data/data/com.termux/files/home/.dub/packages/local-packages.json
Looking for local package map at /data/data/com.termux/files/home/LearnD/.dub/packages/local-packages.json
No valid package found in current working directory: No package file found in /data/data/com.termux/files/home/LearnD/, expected one of dub.json/dub.sdl/package.json
Building package vibe-d in /data/data/com.termux/files/home/.dub/packages/vibe-d-0.9.3/vibe-d/
Refreshing local packages (refresh existing: false)...
Looking for local package map at /var/lib/dub/packages/local-packages.json
Looking for local package map at /data/data/com.termux/files/home/.dub/packages/local-packages.json
Looking for local package map at /data/data/com.termux/files/home/LearnD/.dub/packages/local-packages.json
  Found dependency vibe-d:stream 0.9.3
    Found dependency vibe-d:utils 0.9.3
      Found dependency stdx-allocator 2.77.5
    Found dependency vibe-d:core 0.9.3
      Found dependency vibe-d:data 0.9.3
      Found dependency vibe-core 1.18.0
        Found dependency eventcore 0.9.14
          Found dependency taggedalgebraic 0.11.22
          Found dependency libasync 0.8.6
            Found dependency memutils 1.0.4
  Found dependency vibe-d:web 0.9.3
    Found dependency vibe-d:http 0.9.3
      Found dependency vibe-d:inet 0.9.3
        Found dependency vibe-d:textfilter 0.9.3
      Found dependency vibe-d:tls 0.9.3
        Found dependency openssl 1.1.6+1.0.1g
        Found dependency botan 1.12.19
          Found dependency botan-math 1.0.3
      Found dependency vibe-d:crypto 0.9.3
        Found dependency mir-linux-kernel 1.0.1
      Found dependency diet-ng 1.7.5
  Found dependency vibe-d:mail 0.9.3
  Found dependency vibe-d:redis 0.9.3
  Found dependency vibe-d:mongodb 0.9.3
Refreshing local packages (refresh existing: false)...
Looking for local package map at /var/lib/dub/packages/local-packages.json
Looking for local package map at /data/data/com.termux/files/home/.dub/packages/local-packages.json
Looking for local package map at /data/data/com.termux/files/home/LearnD/.dub/packages/local-packages.json
  Found dependency vibe-d:stream 0.9.3
    Found dependency vibe-d:utils 0.9.3
      Found dependency stdx-allocator 2.77.5
    Found dependency vibe-d:core 0.9.3
      Found dependency vibe-d:data 0.9.3
      Found dependency vibe-core 1.18.0
        Found dependency eventcore 0.9.14
          Found dependency taggedalgebraic 0.11.22
          Found dependency libasync 0.8.6
            Found dependency memutils 1.0.4
  Found dependency vibe-d:web 0.9.3
    Found dependency vibe-d:http 0.9.3
      Found dependency vibe-d:inet 0.9.3
        Found dependency vibe-d:textfilter 0.9.3
      Found dependency vibe-d:tls 0.9.3
        Found dependency openssl 1.1.6+1.0.1g
        Found dependency botan 1.12.19
          Found dependency botan-math 1.0.3
      Found dependency vibe-d:crypto 0.9.3
        Found dependency mir-linux-kernel 1.0.1
      Found dependency diet-ng 1.7.5
  Found dependency vibe-d:mail 0.9.3
  Found dependency vibe-d:redis 0.9.3
  Found dependency vibe-d:mongodb 0.9.3
Generating using build
Running pre-generate commands for vibe-d:tls...
Running echo '
	/+ dub.sdl:
	name "script"
	+/
	import std.algorithm;
	import std.conv;
	import std.file;
	import std.functional;
	import std.path;
	import std.process;
	import std.range;
	import std.stdio;
	import std.string;
	import std.uni;

	void main()
	{
		auto dir = environment.get("DUB_PACKAGE_DIR");
		if (dir.buildPath("tls").exists)  {
			dir = dir.buildPath("tls");
		}
		string opensslVersion;
		try {
			const res = execute(["pkg-config", "openssl", "--modversion"]);
			if (res.status == 0)
				opensslVersion = res.output.strip();
		} catch (Exception e) {}

		if (!opensslVersion.length) try
		{
			const res = execute(["openssl", "version"]).output;
			if (res.canFind("OpenSSL ")) {
				opensslVersion = res.splitter(" ").dropOne.front.filter!(not!(std.uni.isAlpha)).text;
			} else if (res.canFind("LibreSSL ")) {
				writeln("\tWarning: Your default openssl binary points to LibreSSL, which is not supported.");
				version (OSX) {
					writeln("\tOn Mac OSX, this is the default behavior.");
					writeln("\tIf you installed openssl via a package manager, you need to tell DUB how to find it.");
					writeln("\tAssuming brew, run [brew link openssl] and follow the instructions for pkg-config.\n");
				}
			}
		} catch (Exception e) {}
		if (!opensslVersion.length)
		{
			 writeln("\tWarning: Could not find OpenSSL version via pkg-config nor by calling the openssl binary.");
			 writeln("\tAssuming version 1.1.0.");
			 writeln("\tYou might need to export PKG_CONFIG_PATH or install the openssl package if you have a library-only package.");
			 opensslVersion = "1.1.0";
		}
		auto data = text("module openssl_version;\nenum OPENSSL_VERSION=\"", opensslVersion, "\";\n");
		auto output = dir.buildPath("openssl_version.d");
		if (!output.exists || output.readText.strip != data.strip)
			data.toFile(output);
	}
	' | /data/data/com.termux/files/usr/bin/dub - || true
WARNING: linker: /data/data/com.termux/files/usr/tmp/.dub/build/script-~master/application-debug-linux.posix.android-aarch64.arm_hardfloat-ldc_v1.26.0-588CA6B2050D53CB5A61C73BD705F28F/script: unsupported flags DT_FLAGS_1=0x8000001
Configuring dependent vibe-d, deps:"vibe-core", "vibe-d:data", "vibe-d:utils", "vibe-d:crypto", "vibe-d:data", "vibe-d:http", "vibe-d:inet", "vibe-d:mail", "vibe-d:mongodb", "vibe-d:redis", "vibe-d:stream", "vibe-d:textfilter", "vibe-d:tls", "vibe-d:utils", "vibe-d:web"
  Configuring dependent vibe-core, deps:"eventcore", "stdx-allocator"
    Configuring dependent eventcore, deps:"taggedalgebraic"
      Configuring dependent taggedalgebraic, deps:
    Configuring dependent stdx-allocator, deps:
  Configuring dependent vibe-d:data, deps:"vibe-d:utils"
    Configuring dependent vibe-d:utils, deps:"stdx-allocator"
  Configuring dependent vibe-d:crypto, deps:"mir-linux-kernel", "vibe-core", "vibe-d:data", "vibe-d:utils"
    Configuring dependent mir-linux-kernel, deps:
  Configuring dependent vibe-d:http, deps:"diet-ng", "vibe-d:crypto", "vibe-d:inet", "vibe-d:textfilter", "vibe-d:tls"
    Configuring dependent diet-ng, deps:
    Configuring dependent vibe-d:inet, deps:"vibe-core", "vibe-d:data", "vibe-d:utils", "vibe-d:stream", "vibe-d:textfilter"
      Configuring dependent vibe-d:stream, deps:"vibe-core", "vibe-d:data", "vibe-d:utils", "vibe-d:utils"
      Configuring dependent vibe-d:textfilter, deps:"vibe-core", "vibe-d:data", "vibe-d:utils", "vibe-d:utils"
    Configuring dependent vibe-d:tls, deps:"vibe-d:stream"
  Configuring dependent vibe-d:mail, deps:"vibe-d:inet", "vibe-d:tls"
  Configuring dependent vibe-d:mongodb, deps:"vibe-d:http"
  Configuring dependent vibe-d:redis, deps:"vibe-d:http"
  Configuring dependent vibe-d:web, deps:"vibe-d:http"
Performing "debug" build using /data/data/com.termux/files/usr/bin/ldc2 for aarch64, arm_hardfloat.
Target '/data/data/com.termux/files/home/.dub/packages/vibe-d-0.9.3/vibe-d/.dub/build/vibe-core-debug-linux.posix.android-aarch64.arm_hardfloat-ldc_v1.26.0-5FA002B7E01512D3AFECAE75530AA6F6/libvibed.a' doesn't exist, need rebuild.
vibe-d 0.9.3: building configuration "vibe-core"...
Ignoring all import libraries for static library build.
/data/data/com.termux/files/usr/bin/ldc2 -lib -of../.dub/packages/vibe-d-0.9.3/vibe-d/.dub/build/vibe-core-debug-linux.posix.android-aarch64.arm_hardfloat-ldc_v1.26.0-5FA002B7E01512D3AFECAE75530AA6F6/libvibed.a -d-debug -g -w --oq -od=../.dub/packages/vibe-d-0.9.3/vibe-d/.dub/build/vibe-core-debug-linux.posix.android-aarch64.arm_hardfloat-ldc_v1.26.0-5FA002B7E01512D3AFECAE75530AA6F6//obj -d-version=Have_vibe_d -d-version=Have_vibe_d_core -d-version=Have_vibe_core -d-version=Have_vibe_d_data -d-version=Have_vibe_d_utils -d-version=Have_vibe_d_crypto -d-version=Have_vibe_d_http -d-version=Have_vibe_d_inet -d-version=Have_vibe_d_mail -d-version=Have_vibe_d_mongodb -d-version=Have_vibe_d_redis -d-version=Have_vibe_d_stream -d-version=Have_vibe_d_textfilter -d-version=Have_vibe_d_tls -d-version=Have_vibe_d_web -d-version=Have_eventcore -d-version=Have_stdx_allocator -d-version=EventcoreEpollDriver -d-version=Have_taggedalgebraic -d-version=Have_mir_linux_kernel -d-version=Have_diet_ng -d-version=Have_openssl -I../.dub/packages/vibe-d-0.9.3/vibe-d/source/ -I../.dub/packages/vibe-core-1.18.0/vibe-core/source/ -I../.dub/packages/eventcore-0.9.14/eventcore/source/ -I../.dub/packages/taggedalgebraic-0.11.22/taggedalgebraic/source/ -I../.dub/packages/stdx-allocator-2.77.5/stdx-allocator/source/ -I../.dub/packages/vibe-d-0.9.3/vibe-d/data -I../.dub/packages/vibe-d-0.9.3/vibe-d/utils -I../.dub/packages/vibe-d-0.9.3/vibe-d/crypto -I../.dub/packages/mir-linux-kernel-1.0.1/mir-linux-kernel/source/ -I../.dub/packages/vibe-d-0.9.3/vibe-d/http -I../.dub/packages/diet-ng-1.7.5/diet-ng/source/ -I../.dub/packages/vibe-d-0.9.3/vibe-d/inet -I../.dub/packages/vibe-d-0.9.3/vibe-d/stream -I../.dub/packages/vibe-d-0.9.3/vibe-d/textfilter -I../.dub/packages/vibe-d-0.9.3/vibe-d/tls -I../.dub/packages/openssl-1.1.6_1.0.1g/openssl -I../.dub/packages/vibe-d-0.9.3/vibe-d/mail -I../.dub/packages/vibe-d-0.9.3/vibe-d/mongodb -I../.dub/packages/vibe-d-0.9.3/vibe-d/redis -I../.dub/packages/vibe-d-0.9.3/vibe-d/web ../.dub/packages/vibe-d-0.9.3/vibe-d/source/vibe/d.d ../.dub/packages/vibe-d-0.9.3/vibe-d/source/vibe/vibe.d -vcolumns
../.dub/packages/eventcore-0.9.14/eventcore/source/eventcore/drivers/posix/sockets.d(103,21): Error: undefined identifier `MSG_NOSIGNAL`
FAIL ../.dub/packages/vibe-d-0.9.3/vibe-d/.dub/build/vibe-core-debug-linux.posix.android-aarch64.arm_hardfloat-ldc_v1.26.0-5FA002B7E01512D3AFECAE75530AA6F6/ vibed staticLibrary
/data/data/com.termux/files/usr/bin/ldc2 failed with exit code 1.

```

**Expected behavior**
I imagine that package vibe-d must install cleanly without any errors. Installing it for first time.

**Additional information**
$ termux-info
```bash
Packages CPU architecture:
aarch64
Subscribed repositories:
# sources.list
deb https://grimler.se/termux-packages-24/ stable main
# science-repo (sources.list.d/science.list)
deb https://packages.termux.org/apt/termux-science science stable
# game-repo (sources.list.d/game.list)
deb https://packages.termux.org/apt/termux-games games stable
# sources.list.d/pointless.list
deb https://its-pointless.github.io/files/24 termux extras
# root-repo (sources.list.d/root.list)
deb https://packages.termux.org/apt/termux-root root stable
# sources.list.d/pointless.bak
deb https://its-pointless.github.io/files/ termux extras
# x11-repo (sources.list.d/x11.list)
deb https://packages.termux.org/apt/termux-x11 x11 main
Updatable packages:
All packages up to date
Android version:
7.1.2
Kernel build information:
Linux localhost 3.18.31-perf-g89c85bf #1 SMP PREEMPT Tue Oct 22 13:55:14 WIB 2019 aarch64 Android
Device manufacturer:
Xiaomi
Device model:
Redmi Y1


```

**Comment by buttaface** [comment]
@kinke, maybe you can help this guy?

**Comment by kinke** [comment]
> ../.dub/packages/eventcore-0.9.14/eventcore/source/eventcore/drivers/posix/sockets.d(103,21): Error: undefined identifier `MSG_NOSIGNAL`

Might be just a missing definition for Android/Bionic in https://github.com/dlang/druntime/, otherwise https://github.com/vibe-d/eventcore probably needs some Android adaptations . Doesn't have anything to do with the Termux package here.

**Comment by buttaface** [comment]
OK, closing as an upstream bug.

**Comment by Fernal73** [comment]
Ok. So which one of these addresses this issue?

https://github.com/vibe-d/eventcore/search?q=android&type=

Am I to assume that vibe is dead as far as Termux Android is concerned?

**Comment by Grimler91** [comment]
@Fernal73 the [README](https://github.com/vibe-d/eventcore#supported-drivers-and-operating-systems) states partial support for android, consider opening an issue in their repository to check what the status is

**Comment by kinke** [comment]
Looks like this exact issue was fixed in v0.9.16: https://github.com/vibe-d/eventcore/commit/9feabde7ed7c2791b09499bd5e3e5370a00f2dc7
